### PR TITLE
Refactor more network code

### PR DESCRIPTION
**Pull request by ZehMatt**
This PR eliminates the some of the excessive use of std::unique_ptr for the network packets. I also simplified the usage of NetworkPackets a bit, instead of manually having to serialize the command id it is now part of the packet header. I also refactored how network packets are being read, the current implementation reads only the packet size in the first Update and in the next Update it would read the remaining packet data, so a packet would be never fully received with a single Update call, this PR fixes this.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
All seems sensible. Assume compatibility is with the master server? Is that something we should look at fixing.

File: src/openrct2/network/NetworkConnection.cpp [hunk 1]
```diff
@@ -31,90 +32,112 @@ NetworkConnection::~NetworkConnection()
...
+            // If still not enough data for header, keep waiting.
+            return NETWORK_READPACKET_MORE_DATA;
         }
+
+        // Normalise values.
+        header.Size = Convert::NetworkToHost(header.Size);
+        header.Id = ByteSwapBE(header.Id);
+
+        // NOTE: For compatibility reasons we need to remove sizeof(Header.Id) from the size
+        // Previously the Id field was not part of the header rather part of the body
```
[the remark this answers is not in the document]
Comment (ZehMatt) [reply src/openrct2/network/NetworkConnection.cpp:63]: Well yes and older network versions, this one has no network bump as you may have noticed. But I can clarify that this is still required for as long the master server is not being updated.

File: src/openrct2/core/Endianness.h [hunk 2]
```diff
@@ -58,6 +61,22 @@ template<> struct ByteSwapT<8>
 template<typename T> static T ByteSwapBE(const T& value)
 {
     using ByteSwap = ByteSwapT<sizeof(T)>;
-    typename ByteSwap::UIntType result = ByteSwap::SwapBE(reinterpret_cast<const typename ByteSwap::UIntType&>(value));
-    return *reinterpret_cast<T*>(&result);
+    using UIntType = typename ByteSwap::UIntType;
+
+    if constexpr (std::is_enum<T>::value || std::is_integral<T>::value)
```
[the remark this answers is not in the document]
Comment (ZehMatt) [reply src/openrct2/core/Endianness.h:66]: Never knew, good to know.

**Comment by ZehMatt** [comment]
Regarding the master server we should probably have a custom message that is independent from the games network protocol. Updating it may be a bit tricky since it should never break the older versions. I'm thinking about just leaving that one message specific to the master server and refactor that out into a separate file.

**Pull request merged**